### PR TITLE
feat: ability to add operation info as trace or telemetry attributes

### PR DESCRIPTION
**Pull request by SkArchon**
What happens now
* Operation name is part of the expression context, but if you try to manually set it to traces, it will be empty, this is as we evaluate the expression BEFORE setting it to the expression context
* Only Hash / Name / Type is available as attributes

What this PR does
* Better resolving of expressions for traces / telemetry / metrics - Before we set evaluate expressions as before auth and after auth, now we have more steps, like we set the sha256Hash to traces after it has been computed, etc
* Add new operation attributes to expressions

## Summary by CodeRabbit

- **New Features**
  - Added expression-driven telemetry/log attributes for operations: sha256Hash, persistedId, parsingTime, normalizationTime, validationTime, planningTime, and hash.
  - Support blocking mutations based on parsing-time expressions.
  - Improved detection of request.operation.sha256Hash usage for targeted telemetry.

- **Refactor**
  - Centralized, bucketed evaluation of attribute expressions for telemetry, metrics, and tracing.
  - Unified telemetry population flow and streamlined operation timing/identifier propagation.

- **Tests**
  - Large expansion of structured-logging, telemetry, expression-edge-case, visitor, and bucketization tests.

## Checklist

- [ ] I have discussed my proposed changes in an issue and have received approval to proceed.
- [ ] I have followed the coding standards of the project.
- [ ] Tests or benchmarks have been added or updated.
- [ ] Documentation has been updated on [https://github.com/wundergraph/cosmo-docs](https://github.com/wundergraph/cosmo-docs).
- [ ] I have read the [Contributors Guide](https://github.com/wundergraph/cosmo/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SkArchon** [comment]
@coderabbitai review

File: router-tests/telemetry/telemetry_test.go [hunk 1]
```diff
@@ -9593,7 +9598,537 @@ func TestFlakyTelemetry(t *testing.T) {
...
+				require.Len(t, spans, 9)
+
+				detectedSpanCount := validateDetectedSpans(t, spans, key, skipSpans, func(value attribute.Value) {
+					intVal, err := strconv.Atoi(value.AsString())
+					require.NoError(t, err)
+					require.Greater(t, intVal, 0)
+				})
+
+				expected := len(spans) - len(skipSpans)
+				require.Equal(t, expected, detectedSpanCount)
```
[the remark this answers is not in the document]
Comment (SkArchon) [reply router-tests/telemetry/telemetry_test.go:9704]: This is unlikely, and fine since its a test c ase

**Review by alepane21 — APPROVED** [review]
lgtm

**Review by endigma — APPROVED** [review]
